lib/src/test/kotlin/starknet/provider/ProviderTest.kt

```diff
@@ -1200,4 +1200,53 @@ class ProviderTest {
 
         assertNotNull(response)
     }
+
+    @Test
+    fun `batch call contract with block hash and block tag`() {
+        val call1 = Call(
+            contractAddress = balanceContractAddress,
+            entrypoint = "get_balance",
+            calldata = emptyList(),
+        )
+        val blockHash = provider.getBlockHashAndNumber().send().blockHash
+
+        val call2 = Call(
+            contractAddress = balanceContractAddress,
+            entrypoint = "get_balance",
+            calldata = emptyList(),
+        )
+
+        val providerCalls = listOf(
+            provider.callContract(
+                call = call1,
+                blockHash = blockHash,
+            ),
+            provider.callContract(
+                call = call2,
+                blockTag = BlockTag.LATEST,
+            ),
+        )
+        val request = provider.batchRequests(providerCalls)
+        org.junit.jupiter.api.assertAll({ request.send() })
+
+        val response = request.send()
```

Comment:
What is it exactly we are asserting here? Why send another same request right away?

Reply:
In fact my intention was also to test `send` method here, but if so:
1. It should be a separate test (do we actually need it, wdyt?)
2. Usage of `assertAll` is not appropriate here
I will address the change.

Reply:
I don't really understand the first point. Do we need what?

Reply:
I meant, do we also need to test sending `send` method here.

Reply:
Let's make it clear - usage of extra `send` and assertion here was intended to emphasize test of sending the request. As you pointed out, `send` is used anyway, so there's no point to make it redundant and overcomplicate the test. Neither an assertion nor an additional usage of `send` is needed here. They have been removed.